### PR TITLE
Fix aar and aac

### PR DESCRIPTION
**Pull request by sivaramaaa**
This PR speeds up aar !!!!

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -4583,10 +4583,9 @@ static void _anal_calls(RCore *core, ut64 addr, ut64 addr_end) {
 	ut8 buf[4096];
 	ut8 block[4096];
 	bufi = 0;
-	if (addr_end - addr > 0xffffff) {
+	if (addr_end - addr > UT32_MAX) {
```
Comment (radare) [inline libr/core/cmd_anal.c]: not the same

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -4598,14 +4597,17 @@ static void _anal_calls(RCore *core, ut64 addr, ut64 addr_end) {
 		if (!bufi) {
 			r_io_read_at (core->io, addr, buf, sizeof (buf));
 		}
+		memset (block, -1, 4096);
```
Comment (radare) [inline libr/core/cmd_anal.c]: better use sizeof(block) instead of hardcoded 4096 here and there

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -4598,14 +4597,17 @@ static void _anal_calls(RCore *core, ut64 addr, ut64 addr_end) {
...
 		if (!memcmp (buf, block, 4096)) {
 			//eprintf ("Error: skipping uninitialized block \n");
-			break;
+			addr += 4096;
+			continue;
 		}
 		memset (block, 0, 4096);
 		if (!memcmp (buf, block, 4096)) {
```
Comment (radare) [inline libr/core/cmd_anal.c]: use sizeof everywhere. that makes maintainance much easier

File: libr/core/cmd_anal.c (hunk 1 above)
Comment (sivaramaaa) [reply libr/core/cmd_anal.c]: yup , i fixed it , in this binary maxsecW7.exe , as we have merged all sections
from - to range is more than oxffffff , so acc would never analyze that bin at all .

**Comment by radare** [comment]
those sections are fake, shoudlnt be analyzed at all because the contents are empty or overlapped

**Comment by sivaramaaa** [comment]
If we use this function r_core_get_boundaries_prot  , it only returns a single range , since all sections are merged with ur Patch ...

**Pull request merged**